probe-rs/src/architecture/arm/core/armv6m.rs

```diff
@@ -499,11 +499,8 @@ impl<'probe> Armv6m<'probe> {
         })
     }
 
-    fn set_current_core_status(&mut self, status: CoreStatus) {
-        if status != self.state.current_state {
-            self.memory.update_core_status(status);
-            self.state.current_state = status;
-        }
+    fn set_core_status(&mut self, new_status: CoreStatus) {
```

Comment:
I would leave this method completely out of the trait.

Maybe we could make a macro so we do not have to pass all the long arguments, but not sure I like that more :/

Reply:
I'm not sure I understand.

I moved the "updating" logic into `update_core_status` as a freestanding `fn` in `architecture/arm/core/mod.rs`, this is just a shorthand in `impl ArmvN` (no trait impl, just the struct) so we avoid writing the whole thing each time, pretty much like a macro.

Reply:
Oh and to clarify why we even need the bit of logic: it's to avoid spamming the probe with CoreStatus updates that are in reality just "yep we're still in the same state", as that would be pretty wasteful.